File: leetCode/fukun/201-300/lc_219.js
var containsNearbyDuplicate = function (nums, k) {
    for (let i = 0; i < nums.length - 1; i++) {
        for (let j = 1; j <= k; j++) {
            if (i + j >= nums.length) continue;
            if (nums[i] === nums[i + j]) return true;
        }
    }
    return false;
};
console.log(containsNearbyDuplicate([1, 2, 3, 1, 2, 3], 2))
// 以上方法复杂度高
// 可以用map法,单次遍历每一次查找map中对应值的位置，若小于k则返回true,否则覆盖。

File: leetCode/XSS/lc_219.java
import java.util.HashMap;
import java.util.Map;

public class lc_219 {
    public static void main(String[] args) {
        System.out.println(String.valueOf(new Solution().containsNearbyDuplicate(new int[]{1,0,1,1},1)));
    }
    static class Solution {
        public boolean containsNearbyDuplicate(int[] nums, int k) {
            Map<Integer,Integer> map=new HashMap<>();
            for(int i=0;i<nums.length;i++)
            {
                if(map.get(nums[i])==null)
                {
                    map.put(nums[i],i);
                }
                else if(map.get(nums[i])!=null)
                {
                    if(i-map.get(nums[i])<=k)
                    {
                        return true;
                    }
                    else
                    {
                        map.put(nums[i],i);
                    }
                }
            }
            return false;
        }
    }
}
